python/lsst/analysis/ap/apdb.py

```diff
@@ -66,13 +78,65 @@ def connection(self):
         """
         pass
 
-    def load_sources_for_object(self, dia_object_id, limit=100000):
+    def set_bad_DiaSource_flags(self, flag_list):
+        """Set DiaSource flags to exclude.
+
+        Parameters
+        ----------
+        flag_list : `list` [`str`]
+            Flag names to exclude
+        """
+
+        flag_map = os.path.join(lsst.utils.getPackageDir("ap_association"),
+                                "data/association-flag-map.yaml")
+        unpacker = UnpackApdbFlags(flag_map, "DiaSource")
+
+        for flag in flag_list:
+            if flag not in [c[0] for c in unpacker.output_flag_columns['flags']]:
+                raise ValueError(f"flag {flag} not included in DiaSource flags")
+
+        self.bad_DiaSource_flags = flag_list
```

Comment:
As with the method name, I think these are `exclude_source_flags`, not `bad`. And in this context "source" is synonymous with "DiaSource", so no need to spell it out (and it would be `diaSource` anyway here).